### PR TITLE
Lockboxes are now explosion proof

### PR DESCRIPTION
**Pull request by MadmanMartian**
:cl:
 * rscdel: Lockboxes are now explosion proof, and by extension, mecha-drill proof.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
You should include the drilling, not necessarily in the same PR

**Comment by MadmanMartian** [comment]
If I recall, ripley drilling calles ex_act, so this PR would indirectly fix that.

**Comment by ghost** [comment]
>still not emp proof

**Comment by MadmanMartian** [comment]
One change at a time

**Comment by SonixApache** [comment]
downboated because i am a box breaker

and i'd rather see the security team get some sort of warning than have immortal boxes

**Comment by DrSnips** [comment]
Haven't we done the "science shouldn't be able to open weapons" dance before with pomf ~~closing~~ it saying that it shouldn't be merged? Edit: I'm not 100% sure where he said it. But going back to the original lock box stuff he clearly said it was not a change he wanted. In all fairness it was not exactly about blowing them up but rather the results due to opening them improperly.

**Comment by Caboozles** [comment]
This is the worst form of fun police. Just give an sps warning.

**Comment by SonixApache** [comment]
SPS shouldn't be a personal alarm system, tbh. Or should it?

**Comment by Blithering** [comment]
I'd kinda like to see a return of the breaking lock boxes causes weapon damage PR in some form, some of the malfunctions seemed fun

**Comment by aceedex** [comment]
how many times needs the same PR be made

**Comment by Tehsapper** [comment]
why not make them laser-proof as well and give both SPS and station-wide nanotransen update warning

**Comment by DrSnips** [comment]
I just realized something. This is *dab but for coders. The same PR, locked boxes/crates not letting the user just have its contents, keeps getting made over and over along with revert PRs.

**Comment by ShiftyRail** [comment]
I have contradicatory opinions about this

One the one hand, I'm not against tweaking the mechanics of gunsci/guncargo, but I fear this does too much. It would also gives me less opportunities to genocide powergamey scientists, which I will dearly miss

Finally a SPS warning isn't something I feel is needed. SPS should only be about critical emergencies, not potential problems. (SPSes from Black Market were a mistake)

**Comment by utilitas-** [comment]
stop sticking your dick in the SPS as if it needs more functionality, you should only ever get a warning if some other SPS gets nabbed. perfectly reasonable, makes perfect sense, adding anything more just muddles everything, no black market bullshit (at all)
(though I would love to see assembly functionality added, imagine strapping a prox+SPS to the nuke disk and getting a secure BWAAAAARG when someone picks it up (disabled through the use of an EMP, as always))
this would just move the goalposts, like always. if you can't come up with anything interesting, stop fucking fixing that which isn't broken. gun malfunctions would be interesting, at least; this is not.

**Comment by iFlashYou** [comment]
I like how a big ass mech diamond drill should not be able to drill a small box but it can drill open a reinforced wall

**Comment by SonixApache** [comment]
Rather, it shouldn't be able to drill open the box without completely annihilating the contents.

**Comment by iFlashYou** [comment]
Just give it a higher chance to destroy the contents and good is

**Comment by utilitas-** [comment]
And waste 80% more material while doing it. Oh boy, now I just have to tediously print out 10 more laser cannon lockboxes and drill all of them - oh well I came out with more than expected, _better dump the extras on the crew_.
Are you capable of thinking about what you say for even one second?

**Comment by tayachiya** [comment]
You're already ILLEGALLY printing a bunch of guns, should be some kind of risk while doing that.
During blob rounds everyone has all access either way, so I see nothing wrong with breaking stuff

**Comment by DrSnips** [comment]
Then security schould deal with it. Not coders.

**Comment by dsfartgeg** [comment]
Security isn't real, quit trying to convince me otherwise.

Anyway, let's think about this logically:  In real life, ammunition crates are beefy as fuck so that you don't cause an ammo depot explosion by dropping a crate.  They aren't explosion _proof_, just explosion/flame *resistant*.

Maybe make it so they require a bigger boom before opening? Science has TTVs, so it's not like they can't do it at all, but it would require a bit more time and treasure. Maybe a minimum of a welderbomb, which still gives cargo taters options.

**Comment by tayachiya** [comment]
That still leaves a problem of silent drilling into boxes imo

**Comment by dsfartgeg** [comment]
Again, make them tougher: Require drills of sufficient hardness, and make it take forfuckingever and ruin the drill.

**Comment by DrSnips** [comment]
I wouldn't mind a diamond drill being the standard for cracking them.
As for the other part don't pretend like you are going to use a TTV on an rng based lockbox open. Explosions are logged at all bang meters and most bridges now have one which can be read off to the sec channel via a green radio if you really want a way to monitor happenings.

**Comment by SonixApache** [comment]
All explosion-based lockbox openings should just be reported to sec somehow, that would allow and incentivize people to *actually* do something about it.
EMPs, of course, would be exempt from this.

**Comment by ComradeDispenser** [comment]
You know, if you made it need TTV bombs to blow open it would give Toxins more to do...

**Comment by Tehsapper** [comment]
imagine wasting a TTV for a shitty box and not for bombing the armory.

**Comment by dsfartgeg** [comment]
Wasting a TTV?  There's a bomb range, you know.  Nobody will know if you fill it up with lockboxes.

**Comment by utilitas-** [comment]
don't forget:
as many guns as you would ever want is four hacked doors or a spacewalk and two reinforced walls away (on box)
restricting gun production only moves the goalposts, and makes it harder for legitimate use to occur (like miners taking on a xeno queen, or [insert major station threat here]). with lazyrobo's other _magnificent change_, nobody will even have the spare because only a madman or a shitter would actually play captain and getting the spare will be impossible, so no guns, no defending, let's die huddled up in our baby corner while praising our MedRP overlords.

File: code/game/objects/items/weapons/storage/lockbox.dm [hunk 1]
```diff
@@ -126,20 +126,7 @@
...
 
 /obj/item/weapon/storage/lockbox/ex_act(severity)
-	switch(severity)
-		if(1)
-			qdel(src)
-		if(2)
-			if(prob(80))
-				for(var/atom/movable/A in src)
-					remove_from_storage(A, get_turf(src))
-					A.ex_act(3)
```
Comment (Probe1) [inline code/game/objects/items/weapons/storage/lockbox.dm:136]: Look at me mah I did a multi line review just like this popup said

**Comment by Probe1** [comment]
I asked in game too and people voted unanimously against it
```
Vote Result: Close it won with 6 votes.
	Merge it had 0 votes.
```

**Pull request closed without merging**